---
title: "figure-1"
author: "Yaamini Venkataraman"
date: "2025-05-05"
output: html_document
---

In this script, I will create a multipanel figure characterizing the different kinds of metabolomics and lipidomics studies that examine metabolic plasticity in marine invertebrates.

# Set up R Markdown document

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(root.dir = normalizePath(".")) #Set root directory
```

```{r}
getwd()
```

#Install packages

```{r, warnings = FALSE}
#install.packages("tidyverse")
#install.packages("RColorBrewer")
#install.packages("patchwork")
require(tidyverse)
require(RColorBrewer)
require(patchwork)
```

```{r}
sessionInfo()
```

# Import data

```{r}
metadata <- read.csv("../metadata/2025-metabolomics-lipidomics-studies.csv", header = TRUE, col.names = c("Year", "Author", "Title", "Species", "Phyllum", "Stressor", "Method", "dataAcquisition", "Platform", "Type", "statisticalMethods", "enrichmentMethod", "physTraits", "otherOmicMethods", "integrativeAnalysis", "takeHomes", "Abstract", "DOI", "Source")) %>%
  dplyr::select(-c(takeHomes, Abstract, DOI))#Import study information. Change column names to be simple where appropriate. Remove columns that will not be used for figures
head(metadata)
```

# Make figures

## Panel A

Panel A will show the increase in studies over time.

```{r}
panelAPlot <- metadata %>%
  dplyr::select(Year, Author, Method) %>%
  unique(.) %>%
  arrange(Year) %>%
  group_by(Method) %>%
  mutate(cumulative_count = row_number()) %>%
  ggplot(., aes(x = Year, y = cumulative_count, color = Method, linetype = Method)) +
  geom_line(linewidth = 0.75) +
  scale_y_continuous(name = "Cumulative Publications",
                     breaks = c(seq(0,10,5),
                                seq(20,60,20))) +
  scale_x_continuous(limits = c(1990, 2025),
                     breaks = seq(1990, 2025, 5)) +
  scale_color_manual(values = c("orange", "lightskyblue")) +
  scale_linetype_manual(values = c("longdash", "solid")) +
  ggtitle("A. Publishing Trends") +
  theme_classic(base_size = 15) + theme(axis.text.x = element_text(angle = 90,
                                                                   vjust = 0.7))
panelAPlot
```

```{r}
panelAPlot2 <- metadata %>%
  dplyr::select(Year, Author, Method, Phyllum) %>%
  unique(.) %>%
  filter(., Phyllum != "Annelid" & Phyllum != "Brachiopod" & Phyllum != "Bryzoan") %>%
  arrange(Year) %>%
  group_by(Method, Phyllum) %>%
  mutate(cumulative_count = row_number()) %>%
  ggplot(., aes(x = Year, y = cumulative_count, color = Method, linetype = Method)) +
  geom_line(linewidth = 0.75) +
  facet_wrap(~ Phyllum) +
  scale_y_continuous(name = "",
                     breaks = c(seq(0,30,10))) +
  scale_x_continuous(limits = c(1993, 2025),
                     breaks = c(1993, 2025)) +
  scale_color_manual(values = c("orange", "lightskyblue")) +
  scale_linetype_manual(values = c("longdash", "solid")) +
  theme_classic(base_size = 15) + theme(strip.background = element_rect(color = "white"),
                                        axis.text.x = element_text(angle = 90,
                                                                   vjust = 0.7))
panelAPlot2
```

```{r}
panelAFull <- panelAPlot + panelAPlot2 + plot_layout(guides = "collect")
panelAFull
```


## Panel B

I will use a pie chart to breakdown the number of studies that use untargeted, semi-targeted, or targeted data acquisition methods. I will also create a pie chart that examines steady state vs. metabolic flux studies.

```{r}
panelBPlot1 <- metadata %>%
  dplyr::select(Year, Author, dataAcquisition) %>%
  unique(.) %>%
  mutate(., dataAcquisition = gsub(pattern = "\\s*\\([^\\)]+\\)", replacement = "", x = dataAcquisition)) %>%
  mutate(., dataAcquisition = gsub(pattern = ", analyzed only those identified in database", replacement = "", x = dataAcquisition)) %>%
  mutate(., dataAcquisition = gsub(pattern = " ", replacement = "", x = dataAcquisition)) %>%
  group_by(., dataAcquisition) %>%
  summarize(., count = n()) %>%
  ggplot(mapping = aes(x = "", y = count, fill = dataAcquisition)) +
  geom_col(color = "black") +
  coord_polar(theta = "y") +
  scale_fill_manual(name = "",
                    values = brewer.pal(3, "Greys")) +
  ggtitle("B. Data Acquisition Methods") +
  theme_void(base_size = 15)
panelBPlot1
```

```{r}
metadata %>%
  dplyr::select(Year, Author, dataAcquisition) %>%
  unique(.) %>%
  mutate(., dataAcquisition = gsub(pattern = "\\s*\\([^\\)]+\\)", replacement = "", x = dataAcquisition)) %>%
  mutate(., dataAcquisition = gsub(pattern = ", analyzed only those identified in database", replacement = "", x = dataAcquisition)) %>%
  mutate(., dataAcquisition = gsub(pattern = " ", replacement = "", x = dataAcquisition)) %>%
  group_by(., dataAcquisition) %>%
  summarize(., count = n())
```


```{r}
panelBPlot2 <- metadata %>%
  filter(., Method == "Metabolomics") %>%
  dplyr::select(Year, Author, Type) %>%
  unique(.) %>%
  group_by(., Type) %>%
  summarize(., count = n()) %>%
  ggplot(mapping = aes(x = "", y = count, fill = Type)) +
  geom_col(color = "black") +
  coord_polar(theta = "y") +
  scale_fill_manual(name = "",
                    labels = c("31P", "13C", "15N", "Steady state"),
                    values = brewer.pal(4, "Greys")) +
  theme_void(base_size = 15)
panelBPlot2
```

## Panel C

In this panel I want to highlight studies that integrated metabolomics and lipidomics with other methods.

```{r}
metadata %>%
  dplyr::select(Year, Author, Method, otherOmicMethods) %>%
  unique(.) %>%
  pivot_longer(., cols = Method:otherOmicMethods,
               values_to = "Method") %>%
  dplyr::select(-name) %>%
  separate(., col = Method, into = c("Method", "Method2", "Method3"), sep = ", ")
#Only Rodfriguez-Casariego et al. 2024 (Methylation, transcriptome, microbiome, lipidomics) and Wei et al. 2015 (metabolomics, proteomics, and transcriptome) used two methods additional methods
```


```{r}
intStudyTable <- metadata %>%
  dplyr::select(Year, Author, Method, otherOmicMethods) %>%
  unique(.) %>%
  pivot_longer(., cols = Method:otherOmicMethods,
               values_to = "Method") %>%
  dplyr::select(-name) %>%
  separate(., col = Method, into = c("Method", "Method2", "Method3"), sep = ", ") %>%
  pivot_longer(., cols = Method:Method3,
               values_to = "Method") %>%
  dplyr::select(-name) %>%
  filter(., is.na(Method) == FALSE) %>%
  filter(., Method != "None") %>%
  mutate(., Method = gsub(pattern = "16S microbial diversity", replacement = "Microbiome", x = Method)) %>%
  mutate(., Method = gsub(pattern = "16s microbial diversity", replacement = "Microbiome", x = Method)) %>%
  mutate(., Method = gsub(pattern = "Microbiome", replacement = "Microbiome", x = Method)) %>%
  mutate(., Method = gsub(pattern = "ITS2 microbiome", replacement = "Microbiome", x = Method)) %>%
  mutate(., Method = gsub(pattern = "ITS2 sequencing", replacement = "Microbiome", x = Method)) %>%
  mutate(., Method = gsub(pattern = "DNA methylation", replacement = "Epigenomics", x = Method)) %>%
  mutate(., Method = gsub(pattern = "Histone PTM", replacement = "Epigenomics", x = Method)) %>%
  mutate(., Method = gsub(pattern = "transcriptome", replacement = "Transcriptomics", x = Method)) %>%
  mutate(., Method = gsub(pattern = "RT-qPCR", replacement = "Transcriptomics", x = Method)) %>%
  mutate(., used = "Y") %>%
  pivot_wider(., names_from = "Method", values_from = "used", values_fill = "N") %>%
  dplyr::select(Year:Metabolomics, Lipidomics, Transcriptomics, Microbiome, Epigenomics, Proteomics)
#Take metadata and pivot table such that all method information is in one column. Separate entries with multiple methods into different columns, then pivot. Add a column indicating that particular method was used. Pivot dataframe wider, filling in "N" for "NA".
head(intStudyTable)
```

```{r}
intStudySummary <- cbind((intStudyTable %>%
                            filter(., Metabolomics == "Y") %>%
                            dplyr::select(-c(Year, Author)) %>%
                            summarize(., "1metaboliteY" = sum(Metabolomics == "Y"),
                                      "2lipidY" = sum(Lipidomics == "Y"),
                                      "3transcriptY" = sum(Transcriptomics == "Y"),
                                      "5microbiomeY" = sum(Microbiome == "Y"),
                                      "4epigenomeY" = sum(Epigenomics == "Y"),
                                      "6proteinY" = sum(Proteomics == "Y"),
                                      "7noInt" = sum(Lipidomics == "N" & Transcriptomics == "N" & Microbiome == "N" & Epigenomics == "N" & Proteomics == "N")) %>%
                            t(.) %>% as.data.frame(.)),
                         (intStudyTable %>%
                            filter(., Lipidomics == "Y") %>%
                            dplyr::select(-c(Year, Author)) %>%
                            summarize(., "1metaboliteY" = sum(Metabolomics == "Y"),
                                      "2lipidY" = sum(Lipidomics == "Y"),
                                      "3transcriptY" = sum(Transcriptomics == "Y"),
                                      "5microbiomeY" = sum(Microbiome == "Y"),
                                      "4epigenomeY" = sum(Epigenomics == "Y"),
                                      "6proteinY" = sum(Proteomics == "Y"),
                                      "7noInt" = sum(Metabolomics == "N" & Transcriptomics == "N" & Microbiome == "N" & Epigenomics == "N" & Proteomics == "N")) %>%
                            t(.) %>% as.data.frame(.))) #Count the number of metabolomics or lipidomics studies that use other methods, and use cbind to combine.
colnames(intStudySummary) <- c("Metabolomics", "Lipidomics") #Change column names
intStudySummary$Metabolomics[1] <- NA #Change number of metabolomics studies to NA
intStudySummary$Lipidomics[2] <- NA #Change number of lipidomics studies to NA
intStudySummary #Confirm changes
```

```{r}
panelCPlot <- intStudySummary %>%
  rownames_to_column(var = "intMethod") %>%
  pivot_longer(., cols = Metabolomics:Lipidomics, names_to = "method", values_to = "count") %>%
  ggplot(., aes(x = method, y = count, fill = intMethod)) +
  geom_bar(stat = "identity", position = "fill", color = "black") +
  labs(x = "", y = "Percent of Studies") +
  scale_fill_manual(name = "",
                    labels = c("Metabolomics", "Lipidomics",
                               "Transcriptomics", "Epigenomics", "Microbiome", "Proteomics",
                               "None"),
                    values = brewer.pal(7, "Blues")) +
  ggtitle("C. Integrative Molecular Method Used") +
  theme_classic(base_size = 15)
panelCPlot
```

# Combine figures

```{r}
panelAFull / (panelBPlot1 | panelBPlot2) / panelCPlot
ggsave("figure1.pdf", height = 11, width = 8.5)
```

# Count information

I also want to count the number of studies based on various parameters to include in the text.

```{r}
metadata %>%
  dplyr::select(Year, Author, Method, Phyllum) %>%
  unique(.) %>%
  group_by(Phyllum) %>%
  summarize(count = n())
#2 annelid, 1 brachiopod, 1 bryxoan, 17 cnidarian, 13 crustacean, 5 echinoderm, 37 mollusc
```

```{r}
metadata %>%
  dplyr::select(Year, Author, enrichmentMethod) %>%
  unique(.) %>%
  group_by(enrichmentMethod) %>%
  summarize(count = n()) %>%
  View(.)
#10 MetaboAnalyst pathway module
#47 manual mapping
```

```{r}
metadata %>%
  filter(., otherOmicMethods != "None") %>%
  dplyr::select(Year, Author, integrativeAnalysis) %>%
  unique(.) %>%
  group_by(integrativeAnalysis) %>%
  summarize(count = n()) %>%
  View(.)
```

```{r}
metadata %>%
  dplyr::select(Year, Author, physTraits) %>%
  unique(.) %>%
  mutate(., physUsed = case_when(physTraits == "None" ~ "N",
                                 physTraits != "None" ~ "Y")) %>%
  group_by(physUsed) %>%
  summarize(count = n())
#27 no phys, 41 phys
```

```{r}
metadata %>%
  dplyr::select(Year, Author, Source) %>%
  unique(.) %>%
  group_by(Source) %>%
  summarize(count = n())
#Both: 17, Manual: 27, ProQuest: 5, Web of Science: 19
```

